packages/cactus-plugin-keychain-azure-kv/src/test/typescript/unit/webservice-tests.test.ts

```diff
@@ -0,0 +1,210 @@
+import test, { Test } from "tape-promise/tape";
+
+import express from "express";
+import bodyParser from "body-parser";
+import http from "http";
+import { AddressInfo } from "net";
+
+import { IListenOptions, Servers } from "@hyperledger/cactus-common";
+
+import { v4 as uuidv4 } from "uuid";
+
+import { LogLevelDesc } from "@hyperledger/cactus-common";
+
+import {
+  IPluginKeychainAzureKvOptions,
+  AzureCredentialType,
+  PluginKeychainAzureKv,
+} from "../../../main/typescript/public-api";
+
+import {
+  DefaultApi as KeychainAzureKvApi,
+  Configuration,
+} from "../../../main/typescript/generated/openapi/typescript-axios/index";
+
+import fs from "fs";
+import path from "path";
+import os from "os";
+import { PluginRegistry } from "@hyperledger/cactus-core";
+import { SecretClientMock } from "../mock/plugin-keychain-azure-kv-mock";
+
+const logLevel: LogLevelDesc = "TRACE";
+
+test("get,set,has,delete alters state as expected for AzureCredentialType.InMemory", async (t: Test) => {
+  const localStackContainer: IPluginKeychainAzureKvOptions = {
+    instanceId: uuidv4(),
+    keychainId: uuidv4(),
+    logLevel: logLevel,
+    azureEndpoint: "testEndpoint",
+    backend: new SecretClientMock({
+      azureKvUrl: "testUrl",
+      logLevel: logLevel,
+    }),
+  };
+
+  test.onFinish(async () => {
+    await localStackContainer.stop();
+    await localStackContainer.destroy();
+  });
+
+  // Using awsCredentialType: AwsCredentialType.FromAwsCredentialFile
+  {
+    // Create aws credential file in a local directory
+    let tmpDirPath = "tmpDirPath";
+    tmpDirPath = await fs.promises.mkdtemp(path.join(os.tmpdir(), "cactus-"));
+    await fs.promises.writeFile(
+      `${tmpDirPath}/credentials`,
+      "[default]\naws_secret_access_key = test\naws_access_key_id = test",
+      "utf-8",
+    );
+
+    const options1: IPluginKeychainAzureKvOptions = {
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      pluginRegistry: new PluginRegistry({}),
+      azureEndpoint: localstackHost,
+      azureRegion: "us-east-1",
+      azureProfile: "default",
+      azureCredentialType: AzureCredentialType.LocalFile,
+      azureCredentialFilePath: `${tmpDirPath}/credentials`,
+      logLevel: logLevel,
+    };
+    const plugin1 = new PluginKeychainAzureKv(options1);
+
+    const expressApp = express();
+    expressApp.use(bodyParser.json({ limit: "250mb" }));
+    const server = http.createServer(expressApp);
+    const listenOptions: IListenOptions = {
+      hostname: "0.0.0.0",
+      port: 0,
+      server,
+    };
+    const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+    test.onFinish(async () => await Servers.shutdown(server));
+    const { address, port } = addressInfo;
+    const apiHost = `http://${address}:${port}`;
+    const config = new Configuration({ basePath: apiHost });
+    const apiClient = new KeychainAwsSmApi(config);
```

Comment:
@Jeff-Ushry An import is missing that would pull in the DefaultApi class `as KeychainAwsSmApi`